prowler/providers/aws/lib/s3/s3.py

```diff
@@ -147,3 +150,38 @@ def send_to_bucket(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
             )
         return uploaded_objects
+
+    @staticmethod
+    def test_connection(
+        session, bucket_name: str, raise_on_exception: bool = True
+    ) -> Connection:
+        """
+        Test the connection to the S3 bucket.
+
+        Parameters:
+        - session: An instance of the `Session` class representing the AWS session.
+        - bucket_name: A string representing the name of the S3 bucket.
+        - raise_on_exception: A boolean indicating whether to raise an exception if the connection test fails.
+
+        Returns:
+        - A boolean indicating whether the connection to the S3 bucket was successful.
+
+        Raises:
+        - Exception: An exception indicating that the connection test failed.
+        """
+        try:
+            # Set a Temp file to upload
+            with tempfile.TemporaryFile() as temp_file:
+                temp_file.write(b"Test Prowler Connection")
+                temp_file.seek(0)
+                session.upload_fileobj(
```

Comment:
What is the region being used in the client? https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/get_bucket_location.html

To test it use a session from a different region where the bucket belongs because a while ago if bucket does not belong to the same region than the API loads from env (`eu-west-1`) the calls return a 301, causing the process to fail.

Reply:
I've checked the region from the S3 client, if it doesn't match the bucket location I create a new session for the given location.